src/Serialization/main.jl

```diff
@@ -44,136 +44,136 @@ const versionInfo = get_version_info()
 # Right now this is still quite primitive, however there is functionality for
 # encoding Vectors without explicitly adding them to the typeMap. This will
 # have to be adapted though to make it more generic for matrices, arrays, etc.
-const typeMap = Dict{Type, String}([
-    Int => "Base.Int",
-    Graphs.Graph{Graphs.Undirected} => "Oscar.Graphs.Graph{Oscar.Graphs.Undirected}",
-    Graphs.Graph{Graphs.Directed} => "Oscar.Graphs.Graph{Oscar.Graphs.Directed}",
-    SimplicialComplex => "Oscar.SimplicialComplex",
-    Vector => "Vector",
-    Nemo.GaloisField => "Nemo.GaloisField",
-    Nemo.GaloisFmpzField => "Nemo.GaloisFmpzField",
-    fmpz => "fmpz",
-    fmpq => "fmpq",
-    gfp_elem => "Nemo.gfp_elem",
-    gfp_fmpz_elem => "Nemo.gfp_fmpz_elem",
-    fmpq_mpoly => "fmpq_mpoly",
-    fmpz_mpoly => "fmpz_mpoly",
-    fmpq_poly => "fmpq_poly",
-    fmpz_poly => "fmpz_poly",
-    Nemo.NmodRing => "Nemo.NmodRing",
-    Hecke.NfRel{nf_elem} => "Hecke.NfRel{nf_elem}",
-    Polymake.BigObjectAllocated => "Polymake.BigObject",
-    Symbol => "Symbol",
-    FlintRationalField => "FlintRationalField",
-    FmpqPolyRing => "FmpqPolyRing",
-    AbstractAlgebra.Generic.FracField{fmpq_poly} => "AbstractAlgebra.Generic.FracField{fmpq_poly}",
-    AbstractAlgebra.Generic.MPoly{nf_elem} => "AbstractAlgebra.Generic.MPoly{nf_elem}",
-    nf_elem => "nf_elem",
-    AnticNumberField => "AnticNumberField",
-    nmod => "nmod",
-    AbstractAlgebra.Generic.Frac{fmpq_poly} => "AbstractAlgebra.Generic.Frac{fmpq_poly}",
-    AbstractAlgebra.Generic.PolyRing{AbstractAlgebra.Generic.Frac{fmpq_poly}} => "AbstractAlgebra.Generic.PolyRing{AbstractAlgebra.Generic.Frac{fmpq_poly}}",
-    AbstractAlgebra.Generic.MPolyRing{nf_elem} => "AbstractAlgebra.Generic.MPolyRing{nf_elem}",
-    AbstractAlgebra.Generic.Poly{nf_elem} => "AbstractAlgebra.Generic.Poly{nf_elem}",
-    AbstractAlgebra.Generic.PolyRing{nf_elem} => "AbstractAlgebra.Generic.PolyRing{nf_elem}",
-    Vector{AbstractAlgebra.Generic.Poly{nf_elem}} => "Vector{AbstractAlgebra.Generic.Poly{nf_elem}}",
-    NfRelNS{nf_elem} => "NfRelNS{nf_elem}",
-    gfp_poly => "gfp_poly",
-    fq_nmod => "fq_nmod",
-    GFPPolyRing => "GFPPolyRing",
-    FqNmodFiniteField => "FqNmodFiniteField",
-    FqNmodPolyRing => "FqNmodPolyRing",
-    fq_nmod_poly => "fq_nmod_poly",
-    FmpqMPolyRing => "FmpqMPolyRing",
-    Cone{fmpq} => "Cone{fmpq}",
-    Polyhedron{fmpq} => "Polyhedron{fmpq}",
-    PolyhedralComplex{fmpq} => "PolyhedralComplex{fmpq}",
-    UInt64 => "UInt64",
+const typeMap = Dict{Type, String}()
+const reverseTypeMap = Dict{String, Type}()
+
+for (T, str) in (
+    Int8 => "Int8",
+    Int16 => "Int16",
+    Int32 => "Int32",
+    Int64 => "Base.Int",
```

Comment:
Do we really want this to differ from the others?

Of course this might be moot at some point, as we might end up serializing all integers under a single type name, say `Base.Integer` or so, and then upon loading, we deserialize either to a specifically requested integer type, or else to a default one (e.g. `fmpz`)

Reply:
I don't think we want it to be different,  I agree that we probably want to serialize all integers the same and set a default. I believe the Base.Int code was an artifact from https://hackmd.io/s_TcN14VQsOTwii41-XZGg where Lars copied the code into the first working version of serialization